### DISCUSSION TITLE
Questions about Exporting and Conducting Tau-U Models

### DISCUSSION THREAD

**Discussion by gjones1219** [jazznbass/scan#87] [Q&A] [answered]
Hello! Thank you again for this package. I had three questions about Tau-U analyses in the Scan package.

a. I have a single case experiment for which there is no variance in the baseline estimates because all of the baseline measurements are the same. As a result, I get this message:

"Warning message:
In .kendall(A, 1:nA, tau_method = tau_method) :
  could not calculate tau. Variance is zero."

I want to confirm that although this warning comes up, it does not impact my results in any way besides warning me to the non-existent variance in the baseline measures? Alternatively, if this error is more serious for some reason, it would be great to know.

b. Second question: is there any way to export exact p-values for the Tau-U meta analyses using the export function?

c. Third question: is there any way to use the export function to export Tau-U for each single case of a multiple baseline design? This is the least pressing question as I know I can simply calculate the Tau-U for each case in separate models and export each that way, but I wanted to know if there was any way to output these all at once.

Thank you,
Grant

**Comment by gjones1219** [comment]
Following up, thank you!

**Comment by jazznbass** [comment, marked as answer]
Hi Grant,

a. tau-u can not be calculated with zero variance. If you still get results, you should be very carefull in interpreting. Your error code suggests, that you have zero variance in phase A. Therefoe, all calculations including Trend A are not reliable.

b. 
``` r
res <- tau_u(exampleAB)
res$Overall_tau_u
res$Overall_tau_u$p
write.csv(res$Overall_tau_u, "example.csv")
```
c.
just with a workarround (install the package openxlsx before):

```r
library(openxlsx)
res <- tau_u(exampleAB)
write.xlsx(res$table, "test.xlsx")
```

Will result in an Excel file with each case on a separate sheet

I hope, this helps!
Jürgen

**Comment by gjones1219** [comment]
Thank you for this response! Re: the tau-u with zero-variance in phase A, I will definitely make sure to not examine the "trend" outputs, but the output for A vs. B still feels clearly reliable in my case, as the participant experienced no change in scores prior to the intervention, and then experienced much change following. Furthermore, the function did calculate the tau_u with no variance in phase A (I just received a value of "1" for A vs. B, indicating 100% non-overlap between phases). So, for me the clear interpretation is that there is a difference between the two phases. Let me know if there are any issues with this interpretation, otherwise I will plan to use tau-u in this way going forward.

b. and c. sound great. thanks!

Grant
